Madness/SourcePos.swift

```diff
@@ -0,0 +1,49 @@
+//  Copyright (c) 2014 Josh Vera. All rights reserved.
+
+import Prelude
+
+public typealias Line = Int
+public typealias Column = Int
+
+public struct SourcePos<Index: ForwardIndexType>: Equatable {
+
+	public let line: Line
+	public let column: Column
+	public let index: Index
```

Comment:
If we’re parsing some other kind of collection, e.g. an array of tokens, what will the values for `Line`/`Column` be?